### PR TITLE
Add hashed command content to cache file name

### PR DESCRIPTION
**Pull request by rocketspacer**
I need to cache these evals

```
_evalcache pyenv init -
_evalcache pyenv virtualenv-init -
_evalcache pyenv init --path
```

But currently, all these 3 commands generate the same cache file
```
init-pyenv.sh
```

So only 1 of them is cached.

To patch this, I added the md5 hash of the whole command (with args) to the cache file name.

### CODE REVIEW

**Comment by rocketspacer** [comment]
@mroth friendly nudge 👋

File: evalcache.plugin.zsh [hunk 1]
```diff
@@ -7,7 +7,17 @@
 export ZSH_EVALCACHE_DIR=${ZSH_EVALCACHE_DIR:-"$HOME/.zsh-evalcache"}
 
 function _evalcache () {
-  local cacheFile="$ZSH_EVALCACHE_DIR/init-${1##*/}.sh"
+  local cmdHash="nohash"
+
+  # Use md5 on MacOS
+  if command -v md5 > /dev/null; then
```
Comment (mroth) [inline evalcache.plugin.zsh:13]: I believe `if builtin command`... will be a bit more robust from the research I just did (should make sure shell builtin gets used instead of an alias or binary).

File: evalcache.plugin.zsh [hunk 2]
```diff
@@ -7,7 +7,17 @@
 export ZSH_EVALCACHE_DIR=${ZSH_EVALCACHE_DIR:-"$HOME/.zsh-evalcache"}
 
 function _evalcache () {
-  local cacheFile="$ZSH_EVALCACHE_DIR/init-${1##*/}.sh"
+  local cmdHash="nohash"
+
+  # Use md5 on MacOS
```
Comment (mroth) [inline evalcache.plugin.zsh:12]: Recommend removing these comments, they obscure the logic a bit (since we aren't actually checking OS, but rather command presence).

File: evalcache.plugin.zsh [hunk 3]
```diff
@@ -7,7 +7,17 @@
...
+  if command -v md5 > /dev/null; then
+    cmdHash=$(echo -n "$*" | md5)
+  # Fallback to md5sum
+  elif command -v md5sum > /dev/null; then
+    cmdHash=$(echo -n "$*" | md5sum | cut -d' ' -f1)
```
Comment (mroth) [inline evalcache.plugin.zsh:17]: Does `cut` use consistent arg flags across all operating system versions?  (If not, maybe `awk '{print $1}' could work here as an alternative?)

**Review by mroth — COMMENTED** [review]
Thanks for this!  This is an edge case I never thought of.

(This also makes me wish I had a test framework like zunit set up for this, but it's not something I've gotten around to.)

**Comment by memark** [comment]
Also running into this with pyenv. I would be happy if you could follow up on these comments @rocketspacer

File: evalcache.plugin.zsh (hunk 3 above)
Comment (rocketspacer) [reply evalcache.plugin.zsh:17]: I believe it is consistent for the usage I have here.
I'd prefer `cut` over `awk` because it's more widely available.

**Comment by rocketspacer** [comment]
@mroth Address your comments. Also, I've been testing my fork on a personal system for a while, no problem so far.

**Comment by memark** [comment]
Works great! Thanks, @rocketspacer!
